api/src/main/java/org/openmrs/module/reporting/common/SqlIterator.java

```diff
@@ -0,0 +1,116 @@
+package org.openmrs.module.reporting.common;
+
+import org.openmrs.api.APIException;
+import org.openmrs.module.reporting.dataset.DataSetColumn;
+import org.openmrs.module.reporting.dataset.DataSetRow;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.openmrs.module.reporting.dataset.definition.evaluator.SqlFileDataSetEvaluator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqlIterator implements Iterator<DataSetRow> {
```

Comment:
Maybe a better name could be simply `ResultSetIterator`, @mseaton what do you think?